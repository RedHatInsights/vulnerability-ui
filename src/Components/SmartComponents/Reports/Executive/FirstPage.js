import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { Section, Panel, PanelItem } from '@redhat-cloud-services/frontend-components-pdf-generator';
import { DateFormat as dateFormat } from '@redhat-cloud-services/frontend-components/DateFormat';
import { Text } from '@react-pdf/renderer';

import CounterItem from '../Common/CounterItem';
import messages from '../../../../Messages';

import styles from '../Common/styles';

const FirstPage = ({ data,  intl, isEdgeParityEnabled }) => {
    const { totalSystems, totalCves, totalRules, systemCountPerType  } = data;
    const { rpmdnf: conventionalCount, edge: edgeCount } = systemCountPerType || {};

    const shouldShowSystemCounts =  isEdgeParityEnabled && edgeCount > 0;
    return (
        <Fragment key="exec-first-page">
            <Text style={[styles.generated, styles.mb24]}>
                {intl.formatMessage(messages.executiveReportGenerated,
                    {
                        date: dateFormat({ date: new Date(), type: 'exact' }).props.children
                    }
                )}
            </Text>
            <Text style={styles.mb16}>
                {intl.formatMessage(messages.executiveReportHeader)}
            </Text>
            <Text style={styles.mb8}>
                {intl.formatMessage(messages.executiveReportSubHeader, {
                    systems: (
                        <Text style={styles.bold}>
                            {intl.formatMessage(messages.executiveReportHeaderSystems, { systems: totalSystems })}
                        </Text>
                    ),
                    cves: (
                        <Text style={styles.bold}>
                            {intl.formatMessage(messages.executiveReportHeaderCVEs, { cves: totalCves })}
                        </Text>
                    ),
                    rules: (
                        <Text style={styles.bold}>
                            {intl.formatMessage(messages.executiveReportHeaderRules, { rules: totalRules })}
                        </Text>
                    ),
                    hybridCounts: shouldShowSystemCounts ?
                        intl.formatMessage(
                            messages.executiveReportHeaderHybridCounts,
                            {
                                conventionalCount,
                                edgeCount
                            }
                        ) : ''
                })}
            </Text>

            <Section withColumn={false}>
                <Panel>
                    <PanelItem style={styles.counterPanel}>
                        <CounterItem
                            counter={totalSystems}
                            label={intl.formatMessage(messages.executiveReportAnalyzedSystems, { systems: totalSystems })}
                        />
                    </PanelItem>
                    <PanelItem style={styles.counterPanel}>
                        <CounterItem
                            counter={totalCves}
                            label={intl.formatMessage(messages.executiveReportAnalyzedCVEs, { cves: totalCves })}
                        />
                    </PanelItem>
                    <PanelItem style={styles.counterPanel}>
                        <CounterItem
                            counter={totalRules}
                            label={intl.formatMessage(messages.executiveReportAnalyzedRules, { rules: totalRules })}
                        />
                    </PanelItem>
                </Panel>
                {shouldShowSystemCounts ?
                    (<>
                        <Panel>
                            <PanelItem style={styles.counterPanel}>
                                <CounterItem
                                    counter={systemCountPerType?.rpmdnf || 0}
                                    aria-label="Conventional count"
                                    label="Conventional"
                                />
                            </PanelItem>
                        </Panel>
                        <Panel>
                            <PanelItem style={styles.counterPanel}>
                                <CounterItem
                                    counter={systemCountPerType?.edge || 0}
                                    aria-label="Edge count"
                                    label="Immutable"
                                />
                            </PanelItem>
                        </Panel>
                    </>)
                    : null}
            </Section>
        </Fragment>
    );
};

FirstPage.propTypes = {
    data: PropTypes.exact({
        totalSystems: PropTypes.number,
        totalCves: PropTypes.number,
        totalRules: PropTypes.number,
        systemCountPerType: PropTypes.object
    }),
    intl: PropTypes.any,
    isEdgeParityEnabled: PropTypes.bool
};
export default FirstPage;
