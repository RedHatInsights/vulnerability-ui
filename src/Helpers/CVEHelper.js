import { Tooltip } from '@patternfly/react-core';
import { ExternalLinkAltIcon, ServerAltIcon } from '@patternfly/react-icons';
import { processDate } from '@redhat-cloud-services/frontend-components-utilities/files/cjs/helpers';
import React from 'react';
import { BUSINESS_RISK_OPTIONS, STATUS_OPTIONS } from './constants';
import { FormattedMessage } from 'react-intl';
import { InsightsSystemRule, InsightsNoSystemRule } from '../Components/SmartComponents/InsightsSystemRule/InsightsSystemRule';
import messages from '../Messages';

/**
 * Creates the tables rows for systems exposed table
 *
 * @param {Boolean} isLoading
 * @param {Array} payload fetched data
 * @param {Array} openedRows id of opened(expaned) row
 *
 */
export const createExposedSystemsTable = ({ isLoading, payload, openedRows }, cve) => {
    if (isLoading) {
        return { data: [], meta: {}, isLoading };
    }

    let { data, meta } = payload;

    const cvesCount = data && data.length;
    const tableRows = data.map(item => ({
        ...item.attributes,
        id: item.id,
        cve,
        patchAccess: meta.patch_access || false,
        isOpen: openedRows.includes(item.id),
        status: item.attributes.status_name,
        children: item.attributes.rule
            ? <InsightsSystemRule cve={cve} rule={item.attributes.rule} />
            : <InsightsNoSystemRule cve={cve}/>
    }));

    return { data: tableRows, meta: { ...meta, cvesCount  },  isLoading };

};

export function createSystemsListTable(affectedSystems) {
    let isLoading = affectedSystems.isLoading;
    if (!isLoading) {
        let payload = affectedSystems.payload.asMutable();
        let { data, meta } = payload;
        const dataWithStatus = data.map(item => ({
            ...item,
            attributes: {
                ...item.attributes,
                status: (
                    <span>
                        {item.attributes.cve_status_id !== item.attributes.status_id ? (
                            <Tooltip content={<FormattedMessage {...messages.onlyThisSystemCvePair} />}>
                                <ServerAltIcon />
                            </Tooltip>
                        ) : (
                            ''
                        )}{' '}
                        {item.attributes.status_id !== undefined &&
                            STATUS_OPTIONS.find(option => parseInt(option.value) === item.attributes.status_id).label}
                    </span>
                )
            }
        }));
        return { data: dataWithStatus, meta, isLoading };
    }

    return { data: (affectedSystems.payload && affectedSystems.payload.data) || [], meta: {}, isLoading };
}

export function createCveDetailsPage(cves) {
    let isLoading = cves.isLoading;
    let error = cves.error;
    if (!isLoading && !error) {
        let payload = cves.payload.asMutable();
        let { data, meta } = payload;

        data = {
            impact: data.attributes.impact,
            public_date: processDate(data.attributes.public_date),
            description: data.attributes.description,
            synopsis: data.attributes.synopsis,
            rh_link: createRHDBLink(data.attributes.synopsis, messages.RHDBLink),
            mitre_link: createMitreLink(data.attributes.synopsis),
            cvss3_score: data.attributes.cvss3_score,
            cvss2_score: data.attributes.cvss2_score,
            cvss2_metrics: data.attributes.cvss2_metrics,
            cvss3_metrics: data.attributes.cvss3_metrics,
            business_risk: BUSINESS_RISK_OPTIONS.find(
                option => option.value === data.attributes.business_risk_id.toString()
            ).label,
            business_risk_justification: data.attributes.business_risk_text,
            business_risk_id: data.attributes.business_risk_id,
            status: STATUS_OPTIONS.find(option => option.value === data.attributes.status_id.toString()).label,
            status_id: data.attributes.status_id,
            status_justification: data.attributes.status_text,
            systems_status_detail: data.attributes.systems_status_detail,
            systems_status_divergent: data.attributes.systems_status_divergent,
            rules: data.attributes.rules
        };

        return { data, meta, isLoading };
    }

    return { data: [], meta: {}, isLoading, error };
}

export function createRHDBLink(item, link = messages.RHDBLink, props) {
    return (
        <a target="_blank" rel="noopener noreferrer"  href={'https://access.redhat.com/security/cve/' + item} {...props}>
            {<FormattedMessage {...link} />}{<ExternalLinkAltIcon className="pf-u-ml-sm"/>}
        </a>
    );
}

export function createMitreLink(item) {
    return (
        <a target="_blank" rel="noopener noreferrer" href={'https://cve.mitre.org/cgi-bin/cvename.cgi?name=' + item}>
            {<FormattedMessage {...messages.MITRELink} />}{<ExternalLinkAltIcon />}
        </a>
    );
}

export const cveTableRowActions = (methods, rowIndex) => {
    return (rowIndex % 2 === 0) ?
        [
            {
                title: <FormattedMessage {...messages.editBusinessRisk} />,
                onClick: (event, rowId, rowData) =>
                    methods.showBusinessRiskModal([
                        {
                            id: rowData.id,
                            business_risk_id: rowData.business_risk_id,
                            justification: rowData.business_risk_justification
                        }
                    ])
            },
            {
                title: <FormattedMessage {...messages.editStatus} />,
                onClick: (event, rowId, rowData) =>
                    methods.showStatusModal([
                        {
                            id: rowData.id,
                            status_id: rowData.status_id,
                            justification: rowData.status_justification
                        }
                    ])
            }
        ] : null;
};

export const systemCveTableRowActions = (methods, entity, rowIndex) => {
    return (rowIndex % 2 === 0) ?
        [{
            title: <FormattedMessage {...messages.editStatus} />,
            onClick: (event, rowId, rowData) =>
                methods.showStatusModal([
                    {
                        id: rowData.id,
                        display_name: entity.display_name,
                        status_id: rowData.status_id,
                        cve_status_id: rowData.cve_status_id,
                        justification: rowData.status_justification,
                        cve_justification: rowData.cve_status_justification
                    }
                ],
                entity
                )
        }] : null;
};

export const systemExposedTableRowActions = (showStatusModal, cve, rowIndex) => {
    return (rowIndex % 2 === 0) ?
        [{
            title: <FormattedMessage {...messages.editStatus} />,
            onClick: (event, rowId, rowData) => {
                showStatusModal(
                    [{
                        id: cve.id,
                        status_id: cve.status_id,
                        justification: cve.justification
                    }],
                    [{
                        id: rowData.id,
                        display_name: rowData.display_name,
                        status_id: rowData.status_id,
                        justification: rowData.status_text
                    }]
                );
            }
        }] : null;
};

export const systemTableRowActions = (optOut, isOptOut) => [
    {
        title: (
            isOptOut
                ? <FormattedMessage {...messages.systemKebabResumeAnalysis} />
                : <FormattedMessage {...messages.systemKebabExcludeAnalysis} />
        ),
        onClick: (event, rowId, rowData) => {
            optOut([rowData.id]);
        }
    }
];
