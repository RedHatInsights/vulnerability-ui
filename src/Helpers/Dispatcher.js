import { addNotification } from '@red-hat-insights/insights-frontend-components/components/Notifications';
import ReducerRegistry from '../Utilities/ReducerRegistry';
export function dispatchAction(actionCreator) {
    const store = ReducerRegistry.getStore();
    return store.dispatch(actionCreator).catch(res => {
        if ('json' in res) {
            let parsedMessage = res.json();
            parsedMessage.then(arr => {
                for (let err of arr.errors) {
                    store.dispatch(
                        addNotification({
                            variant: 'danger',
                            title: 'Error ' + err.status,
                            description: err.detail
                        })
                    );
                }
            });
        } else {
            store.dispatch(
                addNotification({
                    variant: 'danger',
                    title: 'Error ' + res.status,
                    description: res.statusText
                })
            );
        }
    });
}
