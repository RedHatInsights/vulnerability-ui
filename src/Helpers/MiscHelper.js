import { SecurityIcon, UnknownIcon } from '@patternfly/react-icons';
import { SortByDirection } from '@patternfly/react-table';
import findIndex from 'lodash/findIndex';
import propTypes from 'prop-types';
import React from 'react';
import { impactColorList, impactList, PUBLIC_DATE_OPTIONS } from './constants';
import qs from 'query-string';
import { IntlProvider } from '@redhat-cloud-services/frontend-components-translations';

export const dataShape = propTypes.shape({
    data: propTypes.oneOfType([propTypes.object, propTypes.array]),
    meta: propTypes.object,
    isLoading: propTypes.bool
});

/**
 * Based on the  allowedParams it will construct the arguments for the API call
 * which is defined in the vulnerabilities-client. vulnerabilities-client is written is typescript
 * so we need to define and pass every single parameter. Order in allowedParams is important
 *
 * @param {Object} apiProps
 * @param {Array} allowedParams - order should align with the params in the vulnerabilities-client
 */
export function constructParameters(apiProps, allowedParams) {
    if (apiProps) {
        Object.keys(apiProps).forEach(
            key => (
                apiProps[key] === undefined
                || apiProps[key] === ''
            )
            && delete apiProps[key]
        );

        const params = allowedParams.map(item => apiProps[item]);

        return params;
    }

    return [];
}

// TODO DRY:similar to constructParameters
export function constructURLParameters(urlParams, allowedParams) {
    if (urlParams) {
        const params = { ...urlParams };
        Object.keys(urlParams).forEach(
            key => (
                params[key] === undefined
                || params[key] === ''
                || !allowedParams.includes(key)
                || params[key] === false
            )
            && delete params[key]

        );
        return params;
    }
}

export function formatDate(date = new Date(), includeTime = false) {
    const prepend = (number) => `${`${number}`.length === 1 ? '0' : ''}${number}`;
    const toFormat = new Date(date);
    if (toFormat instanceof Date && !isNaN(toFormat)) {
        const year = toFormat.getFullYear();
        const month = toFormat.getMonth() + 1; // month is zero indexed
        const day = toFormat.getDate();
        const hour = toFormat.getUTCHours();
        const minute = toFormat.getUTCMinutes();

        return includeTime
            ? `${year}-${prepend(month)}-${prepend(day)} ${prepend(hour)}:${prepend(minute)} UTC`
            : `${year}-${prepend(month)}-${prepend(day)}`;
    }

    return toFormat;
}

/**
 * Contructs the filters which have more than one parameter that needs to be passed
 * and resets all the parameteres when it's needed
 *
 * @param {Object} filterParams
 * @return {Object}
 */
export function constructFilterParameters(filterParams) {
    if (Object.prototype.hasOwnProperty.call(filterParams, 'publish_date')) {
        let publicEntry = PUBLIC_DATE_OPTIONS.find(item => item.value === filterParams.publish_date);

        if (typeof publicEntry !== 'undefined') {
            filterParams.public_from = publicEntry.from && formatDate(publicEntry.from);
            filterParams.public_to = publicEntry.to && formatDate(publicEntry.to);
        } else {
            filterParams.public_from = undefined;
            filterParams.public_to = undefined;
            filterParams.publish_date = undefined;
        }
    }

    return filterParams;
}

export function getImpactDetails(impact) {
    if (Object.keys(impactList).includes(impact)) {
        return {
            title: impactList[impact].title,
            color: impactList[impact].color,
            text: impactList[impact].text,
            titleContent: (
                <SecurityIcon size="lg" />
            )
        };
    } else {
        return {
            title: 'Unknown',
            color: impactColorList.unknown,
            text: '',
            titleContent: (
                <UnknownIcon size="lg" />
            )
        };
    }
}

export const createSortBy = (header, value) => {
    if (value) {
        let direction = value.indexOf('-') === 0 ? SortByDirection.desc : SortByDirection.asc;
        value = value.replace(/^(-|\+)/, '');
        const index = findIndex(header, item => item.key === value);
        let sort = {
            index,
            direction
        };
        return sort;
    }

    return {};
};

export const handleSortColumn = (key, direction, header, currentSort, apply) => {
    let columnName = header[key].key;
    const useDefault = currentSort && currentSort.substr(1) !== columnName;
    if (direction === SortByDirection.desc || useDefault) {
        columnName = '-' + columnName;
    }

    apply({ sort: columnName });
};

export const updateStateSet = (stateSet, names, value) => {
    [].concat(names).forEach(name => {
        value ? stateSet.add(name) : stateSet.delete(name);
    });
    return stateSet;
};

export const useUrlParams = (allowedParams) => {
    const url = new URL(window.location);
    const urlParams = qs.parse(url.search);

    const setUrlParams = (parameters) => {
        const searchParams = qs.stringify(constructURLParameters(parameters, allowedParams));
        window.history.replaceState(null, null, `${url.origin}${url.pathname}?${searchParams}`);
    };

    return [urlParams, setUrlParams];
};

export const updateRef = (meta, apply) => {
    const pages = parseInt(meta.pages);
    const page = parseInt(meta.page);
    const cvesCount = parseInt(meta.cvesCount);
    const goTo = (pages === page && page > 1 && cvesCount === 1) ? (pages - 1) : page;

    apply({ page: goTo });
};

export const mountWithIntl = (Component) => {
    const wrapper = mount(Component, {
        wrappingComponent: IntlProvider
    });
    const provider = wrapper.getWrappingComponent();
    provider.setProps({ locale: 'en' });

    return wrapper;
};

export const shallowWithIntl = (Component) => {
    const wrapper = shallow(Component, {
        wrappingComponent: IntlProvider
    });
    const provider = wrapper.getWrappingComponent();
    provider.setProps({ locale: 'en' });

    return wrapper;
};

export const insertIf = (condition, ...elements) => {
    return condition ? elements : [];
};
