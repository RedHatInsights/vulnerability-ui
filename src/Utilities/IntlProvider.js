import { createIntl, createIntlCache } from 'react-intl';

const cache = createIntlCache();
const locale = navigator.language.slice(0, 2);

export const intl = createIntl({
    locale
}, cache);

export const intlSettings = { locale };
