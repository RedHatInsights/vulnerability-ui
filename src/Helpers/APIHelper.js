let BASE_ROUTE = '/api/vulnerability/v1';
import { GitApi } from '@redhat-cloud-services/vulnerabilities-client';
import { SystemProfileApi } from '@redhat-cloud-services/host-inventory-client';
import instance from '../Utilities/interceptors';
import { constructParameters } from './MiscHelper';

let api = new GitApi(undefined, BASE_ROUTE, instance);

const INVENTORY_API_BASE = '/api/inventory/v1';
const systemProfile = new SystemProfileApi(undefined, INVENTORY_API_BASE, instance);

const defaultParams = [
    'filter',
    'limit',
    'offset',
    'page',
    'page_size',
    'sort'
];

export function getCveDetails(synopsis) {
    const result = api.getCveDetails(synopsis);
    return result;
}

export function getSystemDetails(system) {
    const result = api.getSystemDetails(system);
    return result;
}

export function setSystemCveStatus(params) {
    let result = api.setStatus(params);
    return result;
}

export function optOutSystems(systems, optOut) {
    const data = { opt_out: optOut, inventory_id: new Array(...systems) };
    let result = api.setSystemsOptOut(data);
    return result;
}

export function setBusinessRisk(params) {
    let result = api.setCveBusinessRisk(params);
    return result;
}

export function setCveStatus(params) {
    let result = api.setCveStatus(params);
    return result;
}

/**
 * /vulnerabilities/cves
 * /vulnerabilities/cves/ids
 */
let cveParams = [
    'cvss_from',
    'cvss_to',
    'public_from',
    'public_to',
    'impact',
    'data_format',
    'business_risk_id',
    'status_id',
    'rule_presence',
    'tags',
    'sap_sids',
    'sap_system',
    'known_exploit',
    'affecting',
    'rhel_version'
];

export function getCveListByAccount(apiProps) {
    let parameterNames = [
        ...defaultParams,
        ...cveParams,
        'report',
        'advanced_report',
        'ansible',
        'mssql'
    ];
    let parameterArray = constructParameters(apiProps, parameterNames);
    let result = api.getCveList(...parameterArray);
    return result;
}

export function getCveIdsList(apiProps) {
    let parameterNames = [
        ...defaultParams,
        ...cveParams,
        'ansible',
        'mssql'
    ];

    let parameterArray = constructParameters(apiProps, parameterNames);
    let result = api.getCveIdsList(...parameterArray);
    return result;
}

/**
 * /cves/ID/affected_systems
 */
let systemsByCVEparams = [
    'status_id',
    'data_format',
    'uuid',
    'rule_key',
    'rule_presence',
    'rule',
    'tags',
    'sap_sids',
    'sap_system',
    'show_advisories',
    'advisory',
    'rhel_version',
    'first_reported_from',
    'first_reported_to',
    'advisory_available',
    'remediation'
];

export function getAffectedSystemsByCVE({ id, ...apiProps }) {
    let parameterNames = [
        ...defaultParams,
        ...systemsByCVEparams,
        'report',
        'ansible',
        'mssql'
    ];

    let parameterArray = constructParameters(apiProps, parameterNames);
    let result = api.getAffectedSystemsByCve(id, ...parameterArray);
    return result;
}

export function getAffectedSystemsIdsByCve({ id, ...apiProps }) {
    let parameterNames = [
        ...defaultParams,
        ...systemsByCVEparams,
        'ansible',
        'mssql'
    ];

    let parameterArray = constructParameters(apiProps, parameterNames);
    let result = api.getAffectedSystemsIdsByCve(id, ...parameterArray);
    return result;
}

/**
 * /systems/ID/cves
 * /systems/ID/cves/ids
 */
let cveBySystemParams = [
    'cvss_from',
    'cvss_to',
    'public_from',
    'public_to',
    'impact',
    'status_id',
    'data_format',
    'business_risk_id',
    'rule_presence',
    'show_advisories',
    'advisory',
    'rule_key',
    'known_exploit',
    'first_reported_from',
    'first_reported_to',
    'advisory_available',
    'remediation'
];

export function getCveListBySystem(apiProps) {
    const { system } = apiProps;
    let parameterNames = [
        ...defaultParams,
        ...cveBySystemParams,
        'report'
    ];

    if (apiProps && system) {
        Object.keys(apiProps).forEach(key => (apiProps[key] === undefined || apiProps[key] === '') && delete apiProps[key]);
        const params = parameterNames.map(item => apiProps[item]);
        return api.getCveListBySystem(system, ...params).catch(err => {
            if (err && err.status === '404') {
                return { errors: err };
            }

            throw err;
        });
    }
}

export function getCveIdsBySystem(apiProps) {
    const { system } = apiProps;
    let parameterNames = [
        ...defaultParams,
        ...cveBySystemParams
    ];

    if (apiProps && system) {
        Object.keys(apiProps).forEach(key => (apiProps[key] === undefined || apiProps[key] === '') && delete apiProps[key]);
        const params = parameterNames.map(item => apiProps[item]);
        return api.getCveIdsBySystem(system, ...params).catch(err => {
            if (err && err.status === '404') {
                return { errors: err };
            }

            throw err;
        });
    }
}

/**
 * /systems/ids
 * /systems
 */
let systemsParams = [
    'data_format',
    'stale',
    'uuid',
    'tags',
    'sap_sids',
    'sap_system',
    'excluded',
    'rhel_version'
];

export function getSystems(apiProps) {
    let parameterNames = [
        ...defaultParams,
        ...systemsParams,
        'report',
        'ansible',
        'mssql'
    ];
    let parameterArray = constructParameters(apiProps, parameterNames);
    let result = api.getSystemsList(...parameterArray);
    return result;
}

export function getSystemsIds(apiProps) {
    let parameterNames = [
        ...defaultParams,
        ...systemsParams
    ];
    let parameterArray = constructParameters(apiProps, parameterNames);
    let result = api.getSystemsIds(...parameterArray);
    return result;
}

export function getAnnouncement() {
    let result = api.getAnnouncement();
    return result;
}

export function getDashbar(apiProps) {
    let parameterNames = [
        'tags',
        'sap_sids',
        'sap_system',
        'ansible',
        'mssql'
    ];
    let parameterArray = constructParameters(apiProps, parameterNames);
    let result = api.getDashbar(...parameterArray);
    return result;
}

export function getOperatingSystems() {
    return systemProfile.apiSystemProfileGetOperatingSystem();
}

export function fetchPdfReport(template, filters) {
    const CRC_PDF_GENERATE_API = '/api/crc-pdf-generator/v1/generate';
    const url = new URL(CRC_PDF_GENERATE_API,  window.location.origin);

    return fetch(url, {
        method: 'POST',
        headers: {
            'Content-Type': 'application/json'
        },

        body: JSON.stringify({
            service: 'vulnerability',
            template,
            params: {
                ...filters
            }
        })
    }, 50000)
        .then((response) => response.blob());
}
