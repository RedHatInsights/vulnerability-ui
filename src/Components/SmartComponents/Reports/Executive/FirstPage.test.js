import FirstPage from './FirstPage';
import toJson from 'enzyme-to-json';
import { mountWithIntl } from '../../../../Helpers/MiscHelper';
import { BrowserRouter as Router } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { intl } from '../../../../Utilities/IntlProvider';

jest.useFakeTimers('modern').setSystemTime(new Date('2020-01-01').getTime());

const mockStore = configureStore([store => next => action => {}]);
let store = mockStore({ });

const data = {
    "totalSystems": 147,
    "totalCves": 595,
    "totalRules": 11
}

describe('FirstPage component', () => {
    it('Should match snapshot', () => {
        const wrapper = mountWithIntl(
            <Provider store={store}>
                <Router>
                    <FirstPage data={data} intl={intl} />
                </Router>
            </Provider>);        
            
        expect(toJson(wrapper)).toMatchSnapshot();
    })
});
