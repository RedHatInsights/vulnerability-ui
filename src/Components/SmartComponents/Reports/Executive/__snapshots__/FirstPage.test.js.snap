// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FirstPage component Should match snapshot 1`] = `
<Provider
  store={
    {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <FirstPage
        data={
          {
            "totalCves": 595,
            "totalRules": 11,
            "totalSystems": 147,
          }
        }
        intl={
          {
            "$t": [Function],
            "defaultFormats": {},
            "defaultLocale": "en",
            "defaultRichTextElements": undefined,
            "fallbackOnEmptyString": true,
            "formatDate": [Function],
            "formatDateTimeRange": [Function],
            "formatDateToParts": [Function],
            "formatDisplayName": [Function],
            "formatList": [Function],
            "formatListToParts": [Function],
            "formatMessage": [Function],
            "formatNumber": [Function],
            "formatNumberToParts": [Function],
            "formatPlural": [Function],
            "formatRelativeTime": [Function],
            "formatTime": [Function],
            "formatTimeToParts": [Function],
            "formats": {},
            "formatters": {
              "getDateTimeFormat": [Function],
              "getDisplayNames": [Function],
              "getListFormat": [Function],
              "getMessageFormat": [Function],
              "getNumberFormat": [Function],
              "getPluralRules": [Function],
              "getRelativeTimeFormat": [Function],
            },
            "locale": "en",
            "messages": {},
            "onError": [Function],
            "onWarn": [Function],
            "textComponent": Symbol(react.fragment),
            "timeZone": undefined,
          }
        }
      >
        <TEXT
          style={
            [
              {
                "color": "#C9190B",
              },
              {
                "marginBottom": "24px",
              },
            ]
          }
        >
          Report generated 01 Jan 2020 00:00 UTC
        </TEXT>
        <TEXT
          style={
            {
              "marginBottom": "16px",
            }
          }
        >
          This report is an executive summary of vulnerabilities with advisories that may impact your Red Hat Enterprise Linux 

servers.
        </TEXT>
        <TEXT
          style={
            {
              "marginBottom": "8px",
            }
          }
        >
          The vulnerability service is analyzing 
          <TEXT
            key=".1"
            style={
              {
                "fontWeight": "var(--pf-global--FontWeight--bold)",
              }
            }
          >
            147 RHEL systems
          </TEXT>
           and has identified 
          <TEXT
            key=".3"
            style={
              {
                "fontWeight": "var(--pf-global--FontWeight--bold)",
              }
            }
          >
            595 CVEs
          </TEXT>
           and 
          <TEXT
            key=".5"
            style={
              {
                "fontWeight": "var(--pf-global--FontWeight--bold)",
              }
            }
          >
            11 security rules
          </TEXT>
           
that impact 1 or more of these systems.
        </TEXT>
        <Section
          contentStyle={{}}
          style={{}}
          title=""
          titleProps={{}}
          withColumn={false}
        >
          <VIEW
            style={
              {
                "margin": "7 0",
              }
            }
          >
            <TEXT
              style={
                {
                  "color": "#c9190b",
                  "fontSize": 9,
                  "fontWeight": 700,
                  "margin": "7 0",
                }
              }
            />
            <VIEW
              style={{}}
            >
              <Panel
                contentProps={{}}
                style={{}}
                titleProps={{}}
              >
                <VIEW
                  style={{}}
                >
                  <TEXT
                    style={
                      [
                        {
                          "color": "#4f5255",
                          "fontSize": 9,
                          "fontWeight": 700,
                        },
                        {
                          "margin": "7 0",
                        },
                      ]
                    }
                  />
                  <VIEW
                    style={
                      {
                        "display": "flex",
                        "flexDirection": "row",
                        "justifyContent": "space-between",
                      }
                    }
                  >
                    <PanelItem
                      style={
                        {
                          "flex": 6,
                        }
                      }
                      title=" "
                      titleProps={{}}
                    >
                      <VIEW
                        style={
                          {
                            "flex": 6,
                          }
                        }
                      >
                        <TEXT
                          style={
                            {
                              "color": "#6a6e73",
                              "fontSize": 9,
                            }
                          }
                        >
                           
                        </TEXT>
                        <VIEW
                          style={
                            [
                              {
                                "display": "flex",
                                "flexDirection": "row",
                                "justifyContent": "space-between",
                              },
                              {
                                "justifyContent": "flex-start",
                              },
                            ]
                          }
                        >
                          <CounterItem
                            counter={147}
                            label="Analyzed RHEL systems"
                          >
                            <VIEW>
                              <TEXT
                                style={
                                  {
                                    "color": "#C9190B",
                                    "fontSize": 20,
                                  }
                                }
                              >
                                147
                              </TEXT>
                              <TEXT
                                style={
                                  {
                                    "fontSize": 9,
                                  }
                                }
                              >
                                Analyzed RHEL systems
                              </TEXT>
                            </VIEW>
                          </CounterItem>
                        </VIEW>
                      </VIEW>
                    </PanelItem>
                    <PanelItem
                      style={
                        {
                          "flex": 6,
                        }
                      }
                      title=" "
                      titleProps={{}}
                    >
                      <VIEW
                        style={
                          {
                            "flex": 6,
                          }
                        }
                      >
                        <TEXT
                          style={
                            {
                              "color": "#6a6e73",
                              "fontSize": 9,
                            }
                          }
                        >
                           
                        </TEXT>
                        <VIEW
                          style={
                            [
                              {
                                "display": "flex",
                                "flexDirection": "row",
                                "justifyContent": "space-between",
                              },
                              {
                                "justifyContent": "flex-start",
                              },
                            ]
                          }
                        >
                          <CounterItem
                            counter={595}
                            label="Identified CVEs"
                          >
                            <VIEW>
                              <TEXT
                                style={
                                  {
                                    "color": "#C9190B",
                                    "fontSize": 20,
                                  }
                                }
                              >
                                595
                              </TEXT>
                              <TEXT
                                style={
                                  {
                                    "fontSize": 9,
                                  }
                                }
                              >
                                Identified CVEs
                              </TEXT>
                            </VIEW>
                          </CounterItem>
                        </VIEW>
                      </VIEW>
                    </PanelItem>
                    <PanelItem
                      style={
                        {
                          "flex": 6,
                        }
                      }
                      title=" "
                      titleProps={{}}
                    >
                      <VIEW
                        style={
                          {
                            "flex": 6,
                          }
                        }
                      >
                        <TEXT
                          style={
                            {
                              "color": "#6a6e73",
                              "fontSize": 9,
                            }
                          }
                        >
                           
                        </TEXT>
                        <VIEW
                          style={
                            [
                              {
                                "display": "flex",
                                "flexDirection": "row",
                                "justifyContent": "space-between",
                              },
                              {
                                "justifyContent": "flex-start",
                              },
                            ]
                          }
                        >
                          <CounterItem
                            counter={11}
                            label="Identified security rules"
                          >
                            <VIEW>
                              <TEXT
                                style={
                                  {
                                    "color": "#C9190B",
                                    "fontSize": 20,
                                  }
                                }
                              >
                                11
                              </TEXT>
                              <TEXT
                                style={
                                  {
                                    "fontSize": 9,
                                  }
                                }
                              >
                                Identified security rules
                              </TEXT>
                            </VIEW>
                          </CounterItem>
                        </VIEW>
                      </VIEW>
                    </PanelItem>
                    <VIEW
                      style={
                        {
                          "flex": 4,
                        }
                      }
                    >
                      <VIEW
                        style={
                          [
                            {
                              "flexDirection": "row",
                              "justifyContent": "space-evenly",
                            },
                          ]
                        }
                      />
                      <TEXT />
                    </VIEW>
                  </VIEW>
                </VIEW>
              </Panel>
            </VIEW>
          </VIEW>
        </Section>
      </FirstPage>
    </Router>
  </BrowserRouter>
</Provider>
`;
